test/e2e_env/kubernetes/gateway/gatewayapi.go

```diff
@@ -45,22 +45,24 @@ spec:
 			Install(Namespace(externalServicesNamespace)).
 			Install(MTLSMeshKubernetes(meshName)).
 			Install(MeshTrafficPermissionAllowAllKubernetes(meshName)).
-			Install(testserver.Install(
-				testserver.WithName("test-server-1"),
-				testserver.WithMesh(meshName),
-				testserver.WithNamespace(namespace),
-				testserver.WithEchoArgs("echo", "--instance", "test-server-1"),
-			)).
-			Install(testserver.Install(
-				testserver.WithName("test-server-2"),
-				testserver.WithMesh(meshName),
-				testserver.WithNamespace(namespace),
-				testserver.WithEchoArgs("echo", "--instance", "test-server-2"),
-			)).
-			Install(testserver.Install(
-				testserver.WithName("external-service"),
-				testserver.WithNamespace(externalServicesNamespace),
-				testserver.WithEchoArgs("echo", "--instance", "external-service"),
+			Install(Parallel(
```

Comment:
What about using SetupInParallel instead of Parallel in [this line of setup.go](https://github.com/kumahq/kuma/blob/master/test/framework/setup.go#L998)? Do we still need it? Or, is it possible to create `NewClusterSetup` with a default `Parallel` deployment strategy? This would reduce the need for all these changes.

Reply:
The problem is that you need some of it to be sequential. For example, you want to apply namespace and mesh before starting test servers